File: Swift-C++-IR-Comparison/test/controlFlow/Swift/SwitchValueBindingWhere.swift
@main
struct MyMain {

    static func main() {
        let res = switchMe(input: (2, 3))
    }

    static func switchMe(input: (Int, Int)) -> Int {
        var counter = 0
        var max = 7
        switch input {
        case (let x, 0):
            counter = x
        case (0, let y):
            counter = -y
        case let (x, y) where x == y:
            counter = -x
        }
        return counter
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/SwitchValueBindingWhere.cpp

 #include <iostream>

 struct MyMain{

    static int main(){
        int res = switchMe((2, 3));
    }

    static int switchMe(std::pair<int, int> input) {
        int counter = 0;
        int max = 7;
        switch (input.first) {
            case x:
                counter = x;
            case 0:
                counter = -y;
            default: 
                if (x == y){
                    counter = -x;
                }
                else {
                    counter = -1;
                }
        }
        return counter;
    }
 };